v1/plugin/plugin.go

```diff
@@ -302,8 +368,137 @@ type preamble struct {
 	ErrorMessage  string
 }
 
-func startPlugin(srv server, m *meta, p *pluginProxy) int {
-	l, err := net.Listen("tcp", ":"+listenPort)
+func startPlugin(c *cli.Context) error {
+	var (
+		server      *grpc.Server
+		meta        *meta
+		pluginProxy *pluginProxy
+	)
+	libInputOutput.setContext(c)
+	arg, err := getArgs()
+	if err != nil {
+		return err
+	}
+	if lvl := c.Int("log-level"); lvl > arg.LogLevel {
+		log.SetLevel(log.Level(lvl))
+	} else {
+		log.SetLevel(log.Level(arg.LogLevel))
+	}
+	logger := log.WithFields(log.Fields{
+		"_block": "startPlugin",
+	})
+	switch plugin := appArgs.plugin.(type) {
+	case Collector:
+		proxy := &collectorProxy{
+			plugin:      plugin,
+			pluginProxy: *newPluginProxy(plugin),
+		}
+		pluginProxy = &proxy.pluginProxy
+		server, meta, err = buildGRPCServer(collectorType, appArgs.name, appArgs.version, arg, appArgs.opts...)
+		if err != nil {
+			panic(err)
+		}
+		rpc.RegisterCollectorServer(server, proxy)
+	case Processor:
+		proxy := &processorProxy{
+			plugin:      plugin,
+			pluginProxy: *newPluginProxy(plugin),
+		}
+		pluginProxy = &proxy.pluginProxy
+		server, meta, err = buildGRPCServer(processorType, appArgs.name, appArgs.version, arg, appArgs.opts...)
+		if err != nil {
+			panic(err)
+		}
+		rpc.RegisterProcessorServer(server, proxy)
+	case Publisher:
+		proxy := &publisherProxy{
+			plugin:      plugin,
+			pluginProxy: *newPluginProxy(plugin),
+		}
+		pluginProxy = &proxy.pluginProxy
+		server, meta, err = buildGRPCServer(publisherType, appArgs.name, appArgs.version, arg, appArgs.opts...)
+		if err != nil {
+			panic(err)
+		}
+		rpc.RegisterPublisherServer(server, proxy)
+	case StreamCollector:
+		proxy := &StreamProxy{
+			plugin:             plugin,
+			pluginProxy:        *newPluginProxy(plugin),
+			maxCollectDuration: defaultMaxCollectDuration,
+			maxMetricsBuffer:   defaultMaxMetricsBuffer,
+		}
+		pluginProxy = &proxy.pluginProxy
+		server, meta, err = buildGRPCServer(publisherType, appArgs.name, appArgs.version, arg, appArgs.opts...)
+		if err != nil {
+			panic(err)
+		}
+		rpc.RegisterStreamCollectorServer(server, proxy)
+	default:
+		logger.WithField("type", fmt.Sprintf("%T", plugin)).Fatal("Unknown plugin type")
+	}
+
+	if c.Bool("stand-alone") {
+		httpPort := c.Int("stand-alone-port")
+		preamble, err := printPreambleAndServe(server, meta, pluginProxy, arg.ListenPort, arg.Pprof)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+
+		go func() {
+			http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+				fmt.Fprintln(w, preamble)
+			})
+
+			listener, err := net.Listen("tcp", fmt.Sprintf(":%d", httpPort))
+			if err != nil {
+				panic("Unable to get open port")
```

Comment:
Consider using there `log.Fatal(err)` instead of panic, so a user will receive the following message:
```
FATA[0000] listen tcp :8181: bind: address already in use
```